test/rekt/features/apiserversource/data_plane.go

```diff
@@ -822,6 +847,96 @@ func SendsEventsWithRetries() *feature.Feature {
 	return f
 }
 
+func setupNodeLabels(ctxParam context.Context) feature.StepFn {
+	featureFlags := cfgFeat.FromContext(ctxParam)
+	return func(ctx context.Context, t feature.T) {
+		nodes, err := client.Get(ctx).CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Could not list nodes: %v", err)
+		}
+
+		if len(nodes.Items) == 0 {
+			t.Fatal("No nodes found")
+		}
+
+		randomNode := &nodes.Items[0]
+
+		randomNode.Labels = featureFlags.NodeSelector()
+
+		// Update the node with the new labels
+		_, err = client.Get(ctx).CoreV1().Nodes().Update(ctx, randomNode, metav1.UpdateOptions{})
+
+		if err != nil {
+			t.Fatalf("Could not update node: %v", err)
+		}
+	}
+}
+
+func deployAPIServerSauceAndTestLabels(ctxParam context.Context) feature.StepFn {
+	featureFlags := cfgFeat.FromContext(ctxParam)
+	return func(ctx context.Context, t feature.T) {
+		kubeClient := client.Get(ctx)
+
+		source := &v1.ApiServerSource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example-deployment",
+				Namespace: system.Namespace(),
+				UID:       sourceUID,
+			},
+			Spec: v1.ApiServerSourceSpec{
+				Resources: []sourcesv1.APIVersionKindSelector{{
+					APIVersion: "v1",
+					Kind:       "Namespace",
+				}},
+
+				SourceSpec: duckv1.SourceSpec{Sink: sinkDest},
+			},
+		}
+
+		args := resources.ReceiveAdapterArgs{
+			Image:         image,
+			Source:        source,
+			Labels:        resources.Labels(sourceName),
+			SinkURI:       sinkURI.String(),
+			Configs:       &reconcilersource.EmptyVarsGenerator{},
+			Namespaces:    []string{system.Namespace()},
+			AllNamespaces: false,
+			Audience:      &sinkAudience,
+			NodeSelector:  featureFlags.NodeSelector(),
+		}
+
+		dep, err := resources.MakeReceiveAdapter(&args)
+		if err != nil {
+			t.Fatalf("error calling MakeReceiveAdapter: %v", err)
+		}
```

Comment:
Instead of doing this all imperatively here, could you look at how some of the other tests in this file deploy the apiserversource? For example  SendsEventsWithSinkRef would be a good one to look at. You may need to add a helper function in `test/rekt/resources/apiserversource` to set the label, not sure.

Then, you could have in your function creates the feature some steps which install the apiserversource, set things up, etc.

And your assert step would only:
1. get the deployment
2. check that all the pods in the deployment are on the correct node

Currently, you are doing too much for a single step. In general, only install something in a single step, and then verify something about the object you installed in a separate step